### ISSUE TITLE
Add Code Formatting & Linting

### ISSUE THREAD

**Issue by bendzamba** [bendzamba/bookcase#27]
We should adhere to formatting standards for JS, HTML, CSS, JSX, React, Python, SQL, etc.

Ideally we integrate an opinionated, configurable tool, like Black for Python.

Ideally we would also use a pre-commit hook to ensure code formatting before commits.

**Comment by bendzamba** [comment]
I've added black for the backend, and prettier for the frontend. However, I'll close this ticket when I get them both operating automatically with a pre-commit hook.

**Comment by bendzamba** [comment]
This ended up being a medium sized task.

I've switched to ruff for the backend, as it does what black does, but also lints, and seems to be all the rage, primarily for speed (written in Rust).

I finally got eslint running after dealing with some dependency hell. It is sort of tied to prettier's config so hopefully there is no disagreement between the two. 

I believe this will be an iterative thing, where I tweak the config over time to suit my needs, and as I learn more about the interplay between the tools.

I also added the pre-commit package and a global requirements.txt for installing at the root of the project to run from there.

After installing the pre-commit hooks, the three hooks all run on commit.

May be a small issue where, when I use the graphical version control in VSCode, it fails to commit, but from command line it succeeds. Maybe output is too verbose?

Also, need to be vigilant about making sure it's linting/formatting the right files, ideally only the ones that are staged.

It may end up being that running all three on commit is too heavy, but for now I like that it keeps me honest.

Closing with PR.
